### PR TITLE
release.sh: Include haddocks upload for dhall-nix

### PR DESCRIPTION
**Pull request by sjakobi**
Context: https://github.com/dhall-lang/dhall-haskell/issues/1756

### CODE REVIEW

**Comment by sjakobi** [comment]
Note that this is currently untested and my bash skills are rather underdeveloped…

I just got some exercise manually uploading the docs for `hnix` though: https://github.com/haskell-nix/hnix/issues/538#issuecomment-643646729

**Comment by sjakobi** [comment]
Side question: Why is `dhall` the last package that is `release`d in this script? My intuition would have been to put it first to ensure that the Hackage docs builder links the other packages against the latest `dhall` version.

**Comment by Gabriel439** [comment]
@sjakobi: There were two reasons, one of which is no longer applicable:

* Earlier versions of the script wouldn't clean out the directories of release-related files (e.g. `./dist`), so ordering the tree of packages in that way ensured that each package was build against clean versions of its dependencies.  This is no longer necessary as the script cleans up after itself

* Starting with the most downstream packages (e.g. `dhall-lsp-server`) would catch release-related issues as early as possible in the process before additional packages were published, minimizing the "damage"

File: scripts/release.sh [hunk 1]
```diff
@@ -15,6 +15,11 @@ function release {
   cabal v1-configure --disable-tests --disable-benchmarks
   cabal v1-sdist
   cabal upload --publish "dist/${NAME}-${VERSION}.tar.gz"
+  if [ "${NAME}" = "dhall-nix" ]
+  then
+    cabal v2-haddock --builddir=docs --haddock-for-hackage --enable-documentation
```
Comment (Gabriel439) [inline scripts/release.sh:20]: Does `cabal v1-haddock` work for this purpose?  The reason I ask is that the `cabal v1-*` commands work with Cabal's Nix support, which I use on the machine I use to release the packages.  If we use the `cabal v2-*` commands then they need to be run inside of a `nix-shell`

**Comment by sjakobi** [comment]
I hope it will work… You'll find out, @Gabriel439! ;)

**Comment by Gabriel439** [comment]
@sjakobi: Yeah, if anything goes wrong I will submit a follow-up pull request to fix things

**Comment by Gabriel439** [comment]
@sjakobi: I can confirm that it worked :slightly_smiling_face:

**Pull request merged**